### DISCUSSION TITLE
Simplifying Mayo Project Setup with vcpkg

### DISCUSSION THREAD

**Discussion by 181404010226** [fougue/mayo#286] [Ideas]
# Simplifying Mayo Project Setup with vcpkg

Using vcpkg can significantly streamline the installation process for most Mayo dependencies, such as Qt and Assimp. Here's a guide to help you set up the project:

## Install main dependencies using vcpkg:

```
vcpkg install qt5-base:x64-windows
vcpkg install qt5-winextras:x64-windows
vcpkg install qt5-svg:x64-windows
vcpkg install assimp:x64-windows
vcpkg install jemalloc:x64-windows
vcpkg install openvr:x64-windows
vcpkg install ffmpeg:x64-windows
```

> **Note:** FFmpeg installed via vcpkg may lack debug symbols. For full debug capabilities, manual compilation of FFmpeg is recommended. Otherwise, the project can only be run in release mode.

## Install Visual Studio ATL/MFC components:

1. Open Visual Studio Installer
2. Modify your Visual Studio installation
3. In the "Individual components" tab, ensure "C++ ATL for latest v143 build tools (x86 & x64)" and "C++ MFC for latest v143 build tools (x86 & x64)" are selected
4. Apply changes and wait for installation to complete

## OpenCascade installation:

> **Note:** OpenCascade must be installed manually, as vcpkg doesn't provide the necessary env.bat file.

This approach simplifies dependency management for the Mayo project, with the exception of OpenCascade which requires manual installation, and potentially FFmpeg if debug mode is needed.

**Comment by mdemirci-exar** [comment]
Thank you.
